warehouse/api/grpc.go

```diff
@@ -539,11 +539,11 @@ func (g *GRPC) RetryWHUploads(ctx context.Context, req *proto.RetryWHUploadsRequ
 
 func (g *GRPC) CountWHUploadsToRetry(ctx context.Context, req *proto.RetryWHUploadsRequest) (response *proto.RetryWHUploadsResponse, err error) {
 	g.logger.Infow("Count syncs to retry",
-		lf.WorkspaceID, req.WorkspaceId,
-		lf.SourceID, req.SourceId,
-		lf.DestinationID, req.DestinationId,
-		lf.DestinationType, req.DestinationType,
-		lf.IntervalInHours, req.IntervalInHours,
+		obskit.WorkspaceID(req.WorkspaceId),
+		obskit.SourceID(req.SourceId),
+		obskit.DestinationID(req.DestinationId),
+		obskit.DestinationType(req.DestinationType),
+		logger.NewField(lf.IntervalInHours, req.IntervalInHours),
```

Comment:
Why `NewField`? Can't you use a "typed" field instead?

Reply:
It's an `int64` so `logger.NewIntField` would be the best here. `NewField` is like returning `any` basically.

The whole thing about non-sugared logging is that you always infer the type, this way it does not have to use reflection and that's when you really have a benefit in terms of performance.

So TL;DR, always infer the type and use the `n` version of the methods (e.g. `Withn`, `Infon`, `Errorn`, `Warnn`, etc...).

There is no check around this for performance and complexity reasons so you'll have to be aware and do it. The most we could do is build a linter around this.

Reply:
Yeah, we can use NewIntField. Missed it.